src/network/tarcap/packets_handler/handlers/blocks_packet_handler.cpp

```diff
@@ -0,0 +1,57 @@
+#include "blocks_packet_handler.hpp"
+
+#include "dag/dag_block_manager.hpp"
+#include "network/tarcap/packets_handler/syncing_state.hpp"
+
+namespace taraxa::network::tarcap {
+
+BlocksPacketHandler::BlocksPacketHandler(std::shared_ptr<PeersState> peers_state,
+                                         std::shared_ptr<SyncingState> syncing_state,
+                                         std::shared_ptr<DagBlockManager> dag_blk_mgr, const addr_t &node_addr)
+    : PacketHandler(std::move(peers_state), node_addr, "BLOCKS_PH"),
+      syncing_state_(syncing_state),
+      dag_blk_mgr_(dag_blk_mgr) {}
+
+void BlocksPacketHandler::process(const PacketData &packet_data, const dev::RLP &packet_rlp) {
+  std::string received_dag_blocks_str;
+  auto it = packet_rlp.begin();
+  const bool is_final_sync_packet = (*it++).toInt<unsigned>();
+
+  for (; it != packet_rlp.end();) {
+    DagBlock block(*it++);
+    peer_->markBlockAsKnown(block.getHash());
+
+    std::vector<Transaction> new_transactions;
+    for (size_t i = 0; i < block.getTrxs().size(); i++) {
+      Transaction transaction(*it++);
+      new_transactions.push_back(transaction);
```

Comment:
new_transactions.push_back(transaction) could be made more efficient with std::move(transaction)

Reply:
there are two places like this I will fix it